src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeService.java

```diff
@@ -181,9 +181,9 @@ public Dispute setStatus(Long id, DisputeStatus disputeStatus, String rejectedRe
 			throw new NotAllowedException("Unknown status of a Dispute record: %s", dispute.getStatus());
 		}
 
-		dispute.setStatus(disputeStatus);
-		dispute.setRejectedReason(DisputeStatus.REJECTED.equals(disputeStatus) ? rejectedReason : null);
-		return disputeRepository.saveAndFlush(dispute);
+		disputeRepository.setStatus(id, disputeStatus, DisputeStatus.REJECTED.equals(disputeStatus) ? rejectedReason : null);
+		disputeRepository.flushAndClear();
+		return disputeRepository.findById(id).orElseThrow();
```

Comment:
Moved logic to repository so it can get implemented differently by ORDS.
Introduced a flushAndClear so H2 implementation still works.